src/server/main_service.cc

```diff
@@ -753,11 +753,9 @@ Transaction::MultiMode DeduceExecMode(ExecEvalState state,
 
 string CreateExecDescriptor(const std::vector<StoredCmd>& stored_cmds, unsigned num_uniq_shards) {
   string result;
-  result.reserve(stored_cmds.size() * 10);
-  absl::StrAppend(&result, "EXEC/", num_uniq_shards, "\n");
-  for (const auto& scmd : stored_cmds) {
-    absl::StrAppend(&result, "  ", scmd.Cid()->name(), " ", scmd.NumArgs(), "\n");
-  }
+  size_t max_len = std::min<size_t>(20u, stored_cmds.size());
```

Comment:
Sorry, I don't quite follow this logic...
You no longer `StrAppend()` all `stored_cmds` to `result`, so why have this `max_len`? Isn't it confusing, or at least hurts the descriptor in being "unique"?

Reply:
I want to know how long the multi-exec transaction is. most transactions in nature are short so we will track those. we cut the limit to 20 if we track longer transactions. if we omit the length at all we won't know much about transactions

Reply:
What I don't understand is why not **always** use `stored_cmds.size()`, why limit the descriptor to 20?

Reply:
well, we can have "infinite" number of transaction lengths :(
and this descriptor serves as a key to a map. 
we can increase the limit to 1000, I do not mind.